File: ios-native/Sources/PolarisStudyKit/PolarisArea.swift
import Foundation

/// The six areas of Polaris — each fulfilled on-device by a classical algorithm (no AI).
///
/// The desktop/Python build uses an LLM for these; the mobile kit deliberately swaps in
/// deterministic algorithms so it runs instantly, offline, on any device. See `algorithm`.
public enum PolarisArea: String, CaseIterable, Sendable {
    case flashcards
    case quizzing
    case cvBuilder = "cv_builder"
    case advisor
    case citation
    case essay

    /// Human-readable title.
    public var title: String {
        switch self {
        case .flashcards: return "Flashcard Creation"
        case .quizzing: return "Quizzing"
        case .cvBuilder: return "CV Builder"
        case .advisor: return "Advisor"
        case .citation: return "Citation Generator"
        case .essay: return "Essay Helper"
        }
    }

    /// One-line description.
    public var summary: String {
        switch self {
        case .flashcards: return "Turn study material into clear question/answer flashcards."
        case .quizzing: return "Generate quizzes and grade answers with feedback."
        case .cvBuilder: return "Draft and refine a CV / résumé."
        case .advisor: return "General study and academic advice and planning."
        case .citation: return "Produce citations in APA / MLA / Chicago."
        case .essay: return "Outline, draft, and improve essays."
        }
    }

    /// The deterministic algorithm powering this area (no AI).
    public var algorithm: String {
        switch self {
        case .flashcards: return "SM-2 spaced repetition + cloze/definition generation"
        case .quizzing: return "MCQ generation + Levenshtein grading + Leitner boxes"
        case .cvBuilder: return "Template rendering"
        case .advisor: return "Rule-based deadline + spacing scheduler"
        case .citation: return "Rule-table formatter (APA / MLA / Chicago)"
        case .essay: return "Flesch–Kincaid readability + structural heuristics"
        }
    }

    /// Lookup by routing key (matches the Python `PolarisArea` string values).
    public init?(key: String) {
        self.init(rawValue: key.lowercased())
    }
}

File: android-native/polaris-study-kit/src/main/java/com/polaris/studykit/PolarisArea.kt
package com.polaris.studykit

/**
 * The six areas of Polaris — each fulfilled on-device by a classical algorithm (no AI).
 *
 * The desktop/Python build uses an LLM for these; the mobile kits (iOS + Android) deliberately
 * swap in deterministic algorithms so they run instantly, offline, on any device. See [algorithm].
 */
enum class PolarisArea(val key: String, val title: String, val summary: String) {
    FLASHCARDS(
        key = "flashcards",
        title = "Flashcard Creation",
        summary = "Turn study material into clear question/answer flashcards.",
    ),
    QUIZZING(
        key = "quizzing",
        title = "Quizzing",
        summary = "Generate quizzes and grade answers with feedback.",
    ),
    CV_BUILDER(
        key = "cv_builder",
        title = "CV Builder",
        summary = "Draft and refine a CV / résumé.",
    ),
    ADVISOR(
        key = "advisor",
        title = "Advisor",
        summary = "General study and academic advice and planning.",
    ),
    CITATION(
        key = "citation",
        title = "Citation Generator",
        summary = "Produce citations in APA / MLA / Chicago.",
    ),
    ESSAY(
        key = "essay",
        title = "Essay Helper",
        summary = "Outline, draft, and improve essays.",
    ),
    ;

    /** The deterministic algorithm powering this area (no AI). */
    val algorithm: String
        get() = when (this) {
            FLASHCARDS -> "SM-2 spaced repetition + cloze/definition generation"
            QUIZZING -> "MCQ generation + Levenshtein grading + Leitner boxes"
            CV_BUILDER -> "Template rendering"
            ADVISOR -> "Rule-based deadline + spacing scheduler"
            CITATION -> "Rule-table formatter (APA / MLA / Chicago)"
            ESSAY -> "Flesch–Kincaid readability + structural heuristics"
        }

    companion object {
        /** Lookup by routing key (matches the Python/iOS `PolarisArea` string values). */
        fun fromKey(key: String): PolarisArea? = entries.find { it.key == key.lowercase() }
    }
}
